packages/kbn-unified-field-list/src/containers/unified_field_list_sidebar/field_list_sidebar.tsx

```diff
@@ -261,7 +266,7 @@ export const UnifiedFieldListSidebarComponent: React.FC<UnifiedFieldListSidebarP
           onRemoveFieldFromWorkspace={onRemoveFieldFromWorkspace}
           onAddFilter={onAddFilter}
           trackUiMetric={trackUiMetric}
-          multiFields={multiFieldsMap?.get(field.name)} // ideally we better calculate multifields when they are requested first from the popover
+          getMultiFields={useMultiFields ? getMultiFieldsByField : undefined}
```

Comment:
> ideally we better calculate multifields when they are requested first from the popover

Yes, and the reason why I did this, during development the way multi fields were handled triggered an infinite loop, 2 useEffect hooks were playing ping pong. I managed to resolve the problem which was caused by a change in `useGroupedFields` (allFieldsModified was creating a new ref on every hook run), but I thought, let's optimized this, since there was this comment in the code, suggesting it anyway. So the multifields are now just calculated on demand, in the popover. This will improve performance , especially when there's a large field list